generics/services/users.js

```diff
@@ -224,9 +224,55 @@ const profileReadPrivate = function (userId) {
         }
     })
 }
+
+
+
+/**
+  * get subEntities of matching type by recursion.
+  * @method
+  * @name getSubEntitiesBasedOnEntityType
+  * @param parentIds {Array} - Array of entity Ids- for which we are finding sub entities of given entityType
+  * @param entityType {string} - EntityType.
+  * @returns {Array} - Sub entities matching the type .
+*/
+
+async function getSubEntitiesBasedOnEntityType( parentIds, entityType, result ) {
```

Comment:
@Prajwal17Tunerlabs move it to entity service and we dont need to do recursion here please refer this
https://github.com/project-sunbird/ml-core-service/blob/release-4.6.0/module/solutions/helper.js

Reply:
removed getSubEntitiesBasedOnEntityType dependency